docs/sources/shared/wal-data-retention.md

```diff
@@ -0,0 +1,79 @@
+---
+aliases:
+- /docs/agent/shared/wal-data-retention/
+- /docs/grafana-cloud/agent/shared/wal-data-retention/
+- /docs/grafana-cloud/monitor-infrastructure/agent/shared/wal-data-retention/
+- /docs/grafana-cloud/monitor-infrastructure/integrations/agent/shared/wal-data-retention/
+canonical: https://grafana.com/docs/agent/latest/shared/wal-data-retention/
+description: Shared content, information about data retention in the WAL
+headless: true
+---
+
+The `prometheus.remote_write` component reuses the Prometheus Write-Ahead Log
+(WAL) implementation for resiliency against network outages and to avoid the
+prohibitively expensive in-memory buffering of all metric samples. The
+component buffers the received metrics on-disk, in a WAL per configured
```

Comment:
```suggestion
component buffers the received metrics on-disk, in a WAL for each configured
```

Reply:
Isn't the on-disk for the WAL implied? Or can it also be configured/used for.. buffering to RAM (for example)?  Do we need to state "on-disk"?

Reply:
No, it's always on-disk. I wanted to mention the "on-disk" detail at least once in this document in case a reader didn't know. I can leave the reference on the next paragraph and remove this.